pkg/controller/dmaas_backup_controller.go

```diff
@@ -86,6 +89,95 @@ func NewDMaaSBackupController(
 func (d *dmaasBackupController) processBackup(key string) error {
 	log := d.logger.WithField("key", key)
 
-	log.Infof("processing backup")
+	log.Debug("Processing dmaasbackup")
+
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		log.WithError(err).Errorf("failed to split key")
+		return nil
+	}
+
+	original, err := d.lister.DMaaSBackups(ns).Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, "failed to get dmaasbackup")
+	}
+
+	switch original.Spec.State {
+	case v1alpha1.DMaaSBackupStateEmpty, v1alpha1.DMaaSBackupStateActive:
+		// process only active/new dmaasbackup
+		if original.Status.Phase == v1alpha1.DMaaSBackupPhaseCompleted {
+			log.Debug("DMaaSBackup completed, skipping")
+			return nil
+		}
+	case v1alpha1.DMaaSBackupStatePaused:
+		if original.Status.Phase == v1alpha1.DMaaSBackupPhasePaused {
+			log.Debug("DMaaSBackup paused, skipping")
+			return nil
+		}
+		// dmaasbackup is paused but it is not processed by operator
+	}
+
+	dbkp := original.DeepCopy()
+	dbkp.Status.Reason = ""
+
+	logger := logrus.New()
```

Comment:
can we have one time created logger rather than creating it everytime?

Reply:
ok